### PR TITLE
feat: add post for update teams

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] CI is passing

### CODE REVIEW

File: conda_forge_webservices/tests/test_webapp.py [hunk 1]
```diff
@@ -435,3 +435,43 @@ def test_staged_recipes_stale(
...
+                method="POST",
+                body=json.dumps(body),
+                headers=headers,
+            )
+            self.assertEqual(
+                response.code,
+                expected_code,
+                msg=f"token: {token}, feedstock: {feedstock}, hook: {hook}",
+            )
+            if feedstock is not None:
```
Comment (beckermr) [inline conda_forge_webservices/tests/test_webapp.py:472]:
```suggestion
            if feedstock is not None and token is not None:
```

**Comment by isuruf** [comment]
Can you explain what this is doing?

**Comment by beckermr** [comment]
It is part of our security hardening. I am adding a cron-like job [here](https://github.com/conda-forge/admin-migrations/pull/119) to ensure all of the teams are update to date with the proper maintainers. We don't keep tokens with permissions to update teams in github CI secrets (again for security reasons), so I am adding an endpoint here to force the team update when needed.

**Comment by isuruf** [comment]
Then you have to disable commenting in update_teams too, otherwise the bot will spam users who left conda-forge org.

**Comment by beckermr** [comment]
I did. It only comments if the commit hash is not none and I set it to none.

File: conda_forge_webservices/tests/test_webapp.py [hunk 2]
```diff
@@ -435,3 +435,43 @@ def test_staged_recipes_stale(
...
+            self.assertEqual(
+                response.code,
+                expected_code,
+                msg=f"token: {token}, feedstock: {feedstock}, hook: {hook}",
+            )
+            if feedstock is not None and token is not None:
+                update_team_mock.assert_any_call(
+                    "conda-forge",
+                    feedstock,
+                    None,
```
Comment (beckermr) [inline conda_forge_webservices/tests/test_webapp.py:476]:
```suggestion
                    None,  # this sets the commit hash to None so no comments are made
```

Comment (beckermr) [reply conda_forge_webservices/tests/test_webapp.py:476]: cc @isuruf

**Comment by beckermr** [comment]
Any other comments @isuruf or shall I merge?

**Review by isuruf — APPROVED** [review]

**Comment by isuruf** [comment]
We should also use the new github api to get the team by name at https://github.com/conda-forge/conda-smithy/blob/6bdc9e6deb031378e46bf88944f7dacefdf50632/conda_smithy/github.py#L90-L92 to reduce some github api calls

**Comment by beckermr** [comment]
Thanks for the pointer! I'll make an issue for smithy I think.

**Pull request merged**